onnxruntime/core/framework/sequential_executor.cc

```diff
@@ -487,6 +490,65 @@ onnxruntime::Status ExecuteKernel(StreamExecutionContext& ctx,
       }
 #else
       status = p_kernel->Compute(&kernel_ctx);
+
+#if !defined(ORT_MINIMAL_BUILD)
+      auto* node_stats_recorder = ctx.GetSessionState().GetNodeStatsRecorder();
+      if (node_stats_recorder != nullptr) {
+        const auto& node = p_kernel->Node();
+        const OpKernelInfo& op_kernel_info = p_kernel->Info();
+        const auto input_defs = node.InputDefs();
+
+        // Lets first check if any inputs are initializers,
+        // if so we need to account for their memory usage.
+        SafeInt<int64_t> initializers_size = 0;
+        SafeInt<size_t> input_sizes = 0;
+        for (int i = 0, lim = kernel_ctx.InputCount(); i < lim; ++i) {
+          // Need to get ort_value_index for each input.
+          const OrtValue* p_input = kernel_ctx.GetInputMLValue(i);
+          if (p_input != nullptr && p_input->IsAllocated() && p_input->IsTensor()) {
```

Comment:
If some inputs or outputs are external allocated memory using IO binding, shall we include that in accounting?

Reply:
That would be included into accounting since it is thought to be consuming memory on the device. The inputs are not on the device, they would be copied to device anyway.